plugins/modules/rma_workflow_manager.py

```diff
@@ -626,6 +683,37 @@ def validate_device_replacement_params(self):
         self.status = "success"
         return self
 
+    def device_ready_for_replacement_check(self):
+        """
+        Checks if the faulty device is ready for replacement.
+        Parameters:
+            - self (object): An instance of the class that interacts with Cisco Catalyst Center and contains device details.
+        Returns:
+            bool:
+                - True if the faulty device is found and is in the "READY-FOR-REPLACEMENT" state.
+                - False if the faulty device is not found or is not in the "READY-FOR-REPLACEMENT" state.
+        Description:
+            This method retrieves a list of devices marked for replacement from Cisco Catalyst Center
+            using the `device_replacement` API. It iterates through the returned devices to find
+            the specified faulty device based on its serial number, which is stored in the `self.have` attribute.
+            If the faulty device is found and its status is "READY-FOR-REPLACEMENT", the method logs a debug message
+            indicating that the device is already marked for replacement and returns `True`.
+            If the device is not in the "READY-FOR-REPLACEMENT" state or is not found, it returns `False`.
+        """
+        response = self.dnac._exec(
+            family="device_replacement",
+            function='return_replacement_devices_with_details'
+        )
+        devices = response.get("response", [])
+        self.log("Received API response from 'return_replacement_devices_with_details': {0}".format(str(response)), "DEBUG")
+
+        for device in devices:
+            if device.get("faultyDeviceSerialNumber") == self.have.get("faulty_device_serial_number"):
+                if device.get("replacementStatus") == "READY-FOR-REPLACEMENT":
+                    self.log("The device '{0}' is already in the 'MARKED-FOR-REPLACEMENT' state.".format(device.get("faultyDeviceName")), "DEBUG")
```

Comment:
we are checking READY-FOR-REPLACEMENT but printing as MARKED-FOR-REPLACEMENT. Can you check?

Reply:
Updated the code accordingly